rag_experiment/extract.py

```diff
@@ -1,33 +1,103 @@
 # Extract data from Arxiv papers and store them on a disc
 import arxiv
 import tarfile
-
-
+import tempfile
+import gzip
+import os
+import shutil
+import json
 
 # Construct the default API client.
 client = arxiv.Client()
 
-# TODO Function that takes query as an argument for arxiv search and returns the results
-def search_arxiv(query: str) -> arxiv.Search:
-    pass
-
-# Search for the 10 most recent articles matching the keyword "quantum."
-search = arxiv.Search(
-  query = "safety in AI",
-  max_results = 10,
-  sort_by = arxiv.SortCriterion.SubmittedDate
-)
+# Function that takes query as an argument for arxiv search and returns the results
+def search_arxiv(query: str, max_results: int = 10) -> arxiv.Search:
+    search = arxiv.Search(
+        query=query,
+        max_results=max_results,
+        sort_by=arxiv.SortCriterion.SubmittedDate
+    )
+    return client.results(search)
 
-results = client.results(search)
 
-# TODO Download results to a separate folder. Create function that takes results and name of the folder as an input.
+# Download results to a separate folder. Create function that takes results and name of the folder as an input.
 # This should be a temporary folder. 
-def download_results(results: arxiv.Results, folder: str):
-    pass
+def download_results(results: arxiv.Result, temp_dir: str):
+    for paper in results:
+        try:
+            paper.download_source(temp_dir)
+        except Exception as e:
+            print(f"Failed to download {paper.entry_id} due to {e}")  
+
+# Make a list or dictionaries containing metadata from papers
+def extract_metadata(results: arxiv.Result) -> list:
+    metadata_list = []
+    for paper in results:
+        metadata = {
+            'authors': [author.name for author in paper.authors],
+            'title': paper.title,
+            'published': paper.published.strftime('%Y-%m-%d'),
+            'domain': paper.primary_category,
+            'tags': paper.categories
+            }
+        metadata_list.append(metadata)
+    return metadata_list
+        
+        
+# Unpack dowloaded archives, but only .tex files and store them in a separate folder.
+def unpack_archives(source_dir: str, target_dir: str, metadata_list: list):
+    for i, file_name in enumerate(os.listdir(source_dir)):
```

Comment:
```suggestion
    for file_name in os.listdir(source_dir):
```
No need for enumeration. We do not use this `i` anyway..